changelog/unreleased/issue-17450.toml

```
@@ -0,0 +1,19 @@
+type = "changed"
+message = "Dynamically choose default number of process-buffer and output-buffer processors."
```

Let's adjust the wording. I think "dynamically" could be interpreted as "it can change at runtime".

```suggestion
message = "Automatically choose default number of process-buffer and output-buffer processors based on available CPU cores."
```